src/sinks/util/mod.rs

```diff
@@ -104,14 +106,315 @@ impl<I> EncodedEvent<I> {
+                let num_value = field_value_string.parse::<u8>();
+                match num_value {
+                    Ok(num) => {
+                        if num > 23 {
+                            return 1 // USER
+                        } else {
+                            return num
+                        }
+                    }
+                    Err(_) => {
+                            let num = match field_value_string.to_uppercase().as_str() {
+                                "KERN" => 0,
+                                "USER" => 1,
+                                "MAIL" => 2,
+                                "DAEMON" => 3,
+                                "AUTH" => 4,
+                                "SYSLOG" => 5,
+                                "LPR" => 6,
+                                "NEWS" => 7,
+                                "UUCP" => 8,
+                                "CRON" => 9,
+                                "AUTHPRIV" => 10,
+                                "FTP" => 11,
+                                "NTP" => 12,
+                                "SECURITY" => 13,
+                                "CONSOLE" => 14,
+                                "SOLARIS-CRON" => 15,
+                                "LOCAL0" => 16,
+                                "LOCAL1" => 17,
+                                "LOCAL2" => 18,
+                                "LOCAL3" => 19,
+                                "LOCAL4" => 20,
+                                "LOCAL5" => 21,
+                                "LOCAL6" => 22,
+                                "LOCAL7" => 23,
+                                _ => 24,
+                            };
+                            if num > 23 {
+                                return 1 // USER
+                            } else {
+                                return num
+                            }
+                        }
+                    }
+            } else {
+                return 1 // USER
+            }
+        }
+    }
+}
+
+fn get_num_severity(config_severity: &Severity, log: &LogEvent) -> u8 {
+    match config_severity {
+        Severity::Fixed(num) => return *num,
+        Severity::Field(field_name) => {
+            if let Some(field_value) = log.get(field_name.as_str()) {
+                let field_value_string = String::from_utf8(field_value.coerce_to_bytes().to_vec()).unwrap_or_default();
+                let num_value = field_value_string.parse::<u8>();
+                match num_value {
+                    Ok(num) => {
+                        if num > 6 {
+                            return 5 // INFORMATIONAL
+                        } else {
+                            return num
+                        }
+                    }
+                    Err(_) => {
+                            let num = match field_value_string.to_uppercase().as_str() {
```

Comment:
is there some duplication here to deserialization that can be reused?

Reply:
Yes, this and deserialization, for both severity and facility are asking for refactoring.